src/components/Nfts/NftCard.tsx

```diff
@@ -0,0 +1,151 @@
+import {
+  Box,
+  Flex,
+  Image,
+  Skeleton,
+  Tag,
+  TagLeftIcon,
+  Text,
+  useColorModeValue,
+} from '@chakra-ui/react'
+import { useCallback, useMemo, useState } from 'react'
+import Placeholder from 'assets/placeholder.png'
+import PlaceholderDrk from 'assets/placeholder-drk.png'
+import { Amount } from 'components/Amount/Amount'
+import { DiamondIcon } from 'components/Icons/DiamondIcon'
+import { useModal } from 'hooks/useModal/useModal'
+import { getMixPanel } from 'lib/mixpanel/mixPanelSingleton'
+import { MixPanelEvents } from 'lib/mixpanel/types'
+import type { V2ZapperNft } from 'state/apis/zapper/client'
+import { getMediaType } from 'state/apis/zapper/client'
+
+type NftCardProps = {
+  zapperNft: V2ZapperNft
+}
+
+export const NftCard: React.FC<NftCardProps> = ({ zapperNft }) => {
+  const { collection, medias, name, rarityRank } = zapperNft
+  const { floorPriceEth } = collection
+  const mediaUrl = medias?.[0]?.originalUrl
+  const mediaType = getMediaType(mediaUrl)
+  const bg = useColorModeValue('gray.50', 'gray.750')
+  const bgHover = useColorModeValue('gray.100', 'gray.700')
+  const placeholderImage = useColorModeValue(PlaceholderDrk, Placeholder)
+  const [imageLoaded, setImageLoaded] = useState(false)
+
+  const { nft } = useModal()
+
+  const handleClick = useCallback(() => {
+    nft.open({ zapperNft })
+
+    const mixpanel = getMixPanel()
+    const eventData = {
+      name: zapperNft.name,
+      id: zapperNft.id,
+      collectionName: zapperNft.collection.name,
+      collectionAddress: zapperNft.collection.address,
+      estimatedValueEth: zapperNft.estimatedValueEth,
+      collectionFloorPriceEth: zapperNft.collection.floorPriceEth,
+      nftMediaUrls: (zapperNft.medias ?? []).map(media => media.originalUrl),
+    }
+
+    mixpanel?.track(MixPanelEvents.ClickNft, eventData)
+  }, [nft, zapperNft])
+
+  // should take the JSX props below and make them an object instead
+  const mediaBoxProps = useMemo(
```

Comment:
Why memo this? Did we want to action the comment above and make this an object outside of React?